src/main/java/com/stripe/model/CustomerSubscriptionCollection.java

```diff
@@ -12,82 +12,156 @@
 import java.util.Map;
 
 public class CustomerSubscriptionCollection extends StripeCollection<Subscription> {
-  public CustomerSubscriptionCollection list(Map<String, Object> params)
+  // <editor-fold desc="all">
+  /**
+   * List all subscriptions.
+   *
+   * @deprecated Use the {@link #list(Map)} method instead.
+   *     This method will be removed in the next major version.
+   */
+  @Deprecated
+  public CustomerSubscriptionCollection all(Map<String, Object> params)
       throws AuthenticationException, InvalidRequestException,
       APIConnectionException, CardException, APIException {
     return list(params, null);
   }
 
-  public CustomerSubscriptionCollection list(Map<String, Object> params,
-                         RequestOptions options) throws AuthenticationException,
+  /**
+   * List all subscriptions.
+   *
+   * @deprecated Use the {@link #list(Map, RequestOptions)} method instead.
+   *     This method will be removed in the next major version.
+   */
+  @Deprecated
+  public CustomerSubscriptionCollection all(Map<String, Object> params,
+                        RequestOptions options) throws AuthenticationException,
       InvalidRequestException, APIConnectionException, CardException,
       APIException {
-    return APIResource.requestCollection(String.format("%s%s", Stripe.getApiBase(), this.getURL()),
-        params, CustomerSubscriptionCollection.class, options);
+    return list(params, options);
   }
 
+  /**
+   * List all subscriptions.
+   *
+   * @deprecated Use the {@link #list(Map, RequestOptions)} method instead.
+   *     This method will be removed in the next major version.
+   */
   @Deprecated
-  public CustomerSubscriptionCollection all(Map<String, Object> params)
+  public CustomerSubscriptionCollection all(Map<String, Object> params,
+                        String apiKey) throws AuthenticationException,
+      InvalidRequestException, APIConnectionException, CardException,
+      APIException {
+    return list(params, RequestOptions.builder().setApiKey(apiKey).build());
+  }
+  // </editor-fold>
+
+  // <editor-fold desc="create">
+  /**
+   * Create a subscription.
+   *
+   * @deprecated Prefer using the {@link Subscription#create(Map)} method instead.
+   */
+  @Deprecated
+  public Subscription create(Map<String, Object> params)
       throws AuthenticationException, InvalidRequestException,
       APIConnectionException, CardException, APIException {
-    return list(params, null);
+    return create(params, (RequestOptions) null);
   }
 
+  /**
+   * Create a subscription.
+   *
+   * @deprecated Prefer using the {@link Subscription#create(Map, RequestOptions)} method instead.
+   */
   @Deprecated
-  public CustomerSubscriptionCollection all(Map<String, Object> params,
-                        String apiKey) throws AuthenticationException,
+  public Subscription create(Map<String, Object> params,
```

Comment:
I think that would be a much larger change. In this case, we can deprecate the method because subscriptions are a top-level resource with their own endpoints, but that's not the case for all nested list resources (e.g. app fee refunds or transfer reversals).

Reply:
Agreed but since I was reading half of the library in this PR I wanted to flag for the major version